cmd/minishift/cmd/start_preflight.go

```diff
@@ -543,3 +551,12 @@ func checkOriginRelease() bool {
 	}
 	return true
 }
+
+// checkInternetConnectivity return false if user is not connected to internet
+func checkInternetConnectivity() bool {
+	_, err := net.Dial("tcp", "github.com:http")
```

Comment:
this is a utility class, that should be in `pkg/minishift/network/` with parameters to check any host